lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -94,29 +99,251 @@ class IntProgCmdProvider: public CmdProvider {
+ * @return The command type, which is FW_PROG for firmware programming.
+ */
+CMD_TYPE getCmdType() const override 
+{
+    return FW_PROG;
 }
-#endif
+
+    FileManager m_fileManager; /**< Instance of Filemanger class.*/
+
+private:
+    static  CommandInfo m_cmds[]; /**< Array of command information.*/
+    static  ResponseInfo m_responses[]; /**< Array of response information.*/
+    const char* m_fileName = "/zumo_firmware.bin"; /**< Name of the file to be read.*/
+
+    /**
+     * @brief Enumeration of programming commands.
+     * This enumeration defines the various programming commands used during firmware programming.
+     * It includes commands for setting memory addresses and writing data blocks.
+     */
+   enum ProgCmds {
+    FLSPRG_SET_ADDR,    /**< Command to set the memory address. */
+    FLSPRG_WRITE_BLOCK  /**< Command to write a data block to memory. */
+};
+
+    /** 
+    * @brief The next programming command to be executed. 
+    * This variable represents the next programming command to be executed in the sequence.
+    * It initializes with FLSPRG_SET_ADDR, typically used to set the memory address first.
+    */
+    ProgCmds nextCmd = FLSPRG_SET_ADDR;
+    CommandInfo m_updatedCmd; /**<Intermediate storage for updated commands.*/
+    uint8_t* m_updatedCmdBuffer; /**<Buffer for updated commands.*/
+    size_t m_updatedCmdBufferSize; /**<Size of the buffer.*/
+
+    /**
+     * @brief Current memory address for writing data during firmware programming. 
+     * This variable represents the current memory address where data will be written
+     * during the firmware programming process. It is used to keep track of the
+     * memory address for sequential data writes.
+     */
+    uint16_t  m_currWriteMemAddr; 
+
+   /**
+    * @brief Updates the command according to specific requirements.
+    *
+    * This method updates the provided command based on specific requirements
+    * before it is sent for execution. It modifies the command buffer and size
+    * to reflect the changes required by the bootloader communication protocol.
+    * @param fwProgCmd Pointer to the CommandInfo structure representing the command to be updated.
+    * The method modifies the command buffer and size to reflect the changes.
+    * @remarks This method is called before sending each command to the bootloader for execution.
+    * The update process includes modifying command identifiers and parameters to adhere to the
+    * bootloader communication protocol requirements.
+    */
+    void updateFwProgCmd(CommandInfo* fwProgCmd)
+    {
+        if(nullptr != fwProgCmd->command)
+        {   
+            if (fwProgCmd->command == Zumo32U4Specification::SET_MEMORY_ADDR)
+            {
+                /**<Create a buffer for the updated information.*/
+                m_updatedCmdBufferSize = 3;/**< Final Size of the updated command for SET_MEMORY_ADR.*/
+                delete[] m_updatedCmdBuffer; /**<Free existing buffer.*/

```

Comment:
This new/delete is poison for embedded system code. Just use a buffer array that is large enough for the longest command in m_updatedCmdBuffer 128 + x-

Reply:
It will be take care by the next commit.